config/v1/types_infrastructure.go

```diff
@@ -189,6 +192,45 @@ const (
 	IBMCloudProviderTypeUPI IBMCloudProviderType = "UPI"
 )
 
+type CloudControllerManagerState string
+
+const (
+	// Cloud Controller Manager is enabled and expected to be supplied.
+	// Signaling that kubelets and other CCM consumers should use --cloud-provider=external.
+	CloudControllerManagerExternal CloudControllerManagerState = "External"
+
+	// Cloud Controller Manager is enabled and expected to be supplied.
+	// Signaling that kubelets and other CCM consumers should not set --cloud-provider flag.
+	CloudControllerManagerNone CloudControllerManagerState = "None"
+)
+
+type CloudControllerManagerSettings struct {
+	// State determines whether or not an external Cloud Controller Manager is expected to
+	// be presented in the cluster.
+	// For engaging an external Cloud Controller Manager, certain flags are expected to be set to the kubelets.
+	// https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager
+	//
+	// When set to "External", the respective operator (machine config operator) should set `--cloud-provider=external` flag to the kubelet.
+	// When omitted or disabled, no `cloud-provider` flag should be set.
+	// +kubebuilder:validation:Enum=External;None
+	// +optional
+	State CloudControllerManagerState `json:"state,omitempty"`
```

Comment:
Rather than omitempty, allow `""` in the enum and then this will improve discoverability as the field will be shown with an empty value. That or we want to default it

Reply:
Updated, did use the language which you proposed for const docs in prior comments above.